flang/test/Driver/isysroot.f90

```diff
@@ -0,0 +1,12 @@
+! Verify that the -isysroot flag is known to the frontend and, on Darwin,
+! is passed on to the linker.
+
+! RUN: %flang -### --target=aarch64-apple-darwin -isysroot /path/to/sysroot \
+! RUN:        %s 2>&1 | FileCheck %s --check-prefix=CHECK-DARWIN
+! RUN: %flang -### --target=aarch64-linux-gnu -isysroot /path/to/sysroot \
+! RUN:        %s 2>&1 | FileCheck %s --check-prefix=CHECK-LINUX
+
+! CHECK-DARWIN: "{{.*[\/]}}ld" {{.*}}"-syslibroot" "/path/to/sysroot"
+! Unused on Linux.
+! CHECK-LINUX: warning: argument unused during compilation: '-isysroot /path/to/sysroot'
```

Comment:
This doesn't feel right - why is it ignored on Linux?

Reply:
With clang the behavior of `-isysroot` is also different on Darwin. The man page of gcc explains it better:
```
    This option is like the --sysroot option, but applies only to
    header files (except for Darwin targets, where it applies to
    both header files and libraries).
```

As Fortran doesn't use header files (at least not by default), `-isysroot` ends up unused on other targets.

I guess we could use `--sysroot` instead, but I wanted to match what our build system does when building C/C++ sources with clang, which is use `-isysroot` on Darwin.

Reply:
Thank you for the explanation - I think that it would be really helpful to expand the docs a bit more. Also, given that `-isysroot` is critical for generating executables on Darwin, it would be great to test exactly that. But not sure how to achieve that without knowing the location of system libraries on Darwin :/

Reply:
There is actually a way to test generating executables on Darwin, by using the sysroot detected by CMake. I've added this test to `exec-darwin.f90`.
`FlangDriver.md` has been expanded with your suggestions. I guess that is enough for the docs, right?